### PR TITLE
update readme && add statefulset workload

### PR DESCRIPTION
**Pull request by wonderflow**
add trait-injector

### CODE REVIEW

File: traits/README.md [hunk 1]
```diff
@@ -4,8 +4,13 @@ Currently we have OAM traits as below:
 
 ## Core Traits
 
-- [ManualScaler](https://github.com/crossplane/addon-oam-kubernetes-local)
+- [ManualScaler](https://github.com/crossplane/addon-oam-kubernetes-local) can enable manually scale replica to workload.
```
Comment (zzxwill) [inline traits/README.md:7]:
```suggestion
- [ManualScaler](https://github.com/crossplane/addon-oam-kubernetes-local) can enable replicas of workloads to be manually scaled.
```

File: workloads/README.md [hunk 2]
```diff
@@ -4,6 +4,6 @@ Currently we have OAM workloads as below:
 
 ## Core Workloads
 
-- [ContainerizedWorkload](https://github.com/crossplane/addon-oam-kubernetes-local)
+- [ContainerizedWorkload](https://github.com/crossplane/addon-oam-kubernetes-local) is core workload implementation of OAM.
```
Comment (zzxwill) [inline workloads/README.md:7]:
```suggestion
- [ContainerizedWorkload](https://github.com/crossplane/addon-oam-kubernetes-local) is the core workload implementation of OAM.
```

File: workloads/statefulset/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# StatefulSet Workload
+
+oam-kubernetes-runtime also support K8s native resource, this section will introduce how to use K8s StatefulSet.
```
Comment (zzxwill) [inline workloads/statefulset/README.md:3]:
```suggestion
oam-kubernetes-runtime also supports K8s native resource, this section will introduce how to use K8s StatefulSet.
```

File: workloads/statefulset/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+## Update RBAC for OAM AppConfig Controller
+
+Make sure your role has below rules:
```
Comment (zzxwill) [inline workloads/statefulset/README.md:7]: How about changing `rules` to `clusterrole` as they are different K8s primitives :)

File: workloads/statefulset/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+## Run AppConfig
+
+```shell script
+$ kubectl apply  -f sampeapp.yaml
+applicationconfiguration.core.oam.dev/example-appconfig created
+```
+
+Then you will see a StatuefulSet app has been created.
```
Comment (zzxwill) [inline workloads/statefulset/README.md:37]:
```suggestion
Then you will see a StatuefulSet app which has been created.
```

File: workloads/statefulset/sampeapp.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: example-appconfig
+spec:
+  components:
+    - componentName: example-sts
```
Comment (zzxwill) [inline workloads/statefulset/sampeapp.yaml:7]: I saw the warning `No newline at end of file`, so how about adding new a new line in this file and those blow :)

File: workloads/statefulset/README.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+rules:
+- apiGroups:
+  - apps
+  resources:
+  - statefulsets
+  verbs:
+  - *
+```
+
+## Install Component
```
Comment (zzxwill) [inline workloads/statefulset/README.md:23]: Do we need to say `please enter folder workloads/statefulset` or `cd workloads/statefulset` before we execute `kubectl apply`?

File: workloads/statefulset/README.md (hunk 4 above)
Comment (wonderflow) [reply workloads/statefulset/README.md:7]: Of course, thanks

File: workloads/statefulset/sampeapp.yaml (hunk 6 above)
Comment (wonderflow) [reply workloads/statefulset/sampeapp.yaml:7]: that's not important, but I'm fine

File: workloads/statefulset/README.md (hunk 5 above)
Comment (wonderflow) [reply workloads/statefulset/README.md:37]: I think the old one is also right

File: workloads/statefulset/README.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
+# StatefulSet Workload
+
+oam-kubernetes-runtime also supports K8s native resource, this section will introduce how to use K8s StatefulSet.
+
+## Update RBAC for OAM AppConfig Controller
+
+Make sure your cluster-role has following rules:
+
+```yaml
```
Comment (zzxwill) [inline workloads/statefulset/README.md:9]: Do you plan to put this `ClusterRole` yaml to role.yaml?

Comment (wonderflow) [reply workloads/statefulset/README.md:9]: That need to change crossplane repo which is out of this PR's scope.

File: workloads/statefulset/README.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+kind: ClusterRole
+metadata:
+  name: <your-cluster-role>
+rules:
+- apiGroups:
+  - apps
+  resources:
+  - statefulsets
+  verbs:
+  - *
```
Comment (zzxwill) [inline workloads/statefulset/README.md:20]:
```suggestion
  - "*"
```

File: workloads/statefulset/sample-sts-component.yaml [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+        matchLabels:
+          app: test
+      template:
+        metadata:
+          labels:
+            app: test
+        spec:
+          containers:
+            - name: wordpress
+              image: wordpress:4.6.1-apache
```
Comment (zzxwill) [inline workloads/statefulset/sample-sts-component.yaml:23]: Verified the readiness of resource StatefulSet by changing the container.
```suggestion
            - name: nginx
              image: nginx:1.17
```

**Pull request merged**